BUG: build failure on PyPy in hash functions

The PyPy builds on conda-forge are failing with 

```
    In file included from pandas/_libs/algos.c:663:
    pandas/_libs/src/klib/khash_python.h: In function 'complexobject_hash':
    pandas/_libs/src/klib/khash_python.h:282:38: error: '_PyHASH_IMAG' undeclared (first use in this function)
      282 |     Py_uhash_t combined = realhash + _PyHASH_IMAG * imaghash;
          |                                      ^~~~~~~~~~~~
    pandas/_libs/src/klib/khash_python.h:282:38: note: each undeclared identifier is reported only once for each function it appears in
```

This seems to be related to https://github.com/pandas-dev/pandas/pull/41988 cc @realead

--- a/doc/source/whatsnew/v1.3.1.rst
+++ b/doc/source/whatsnew/v1.3.1.rst
@@ -14,7 +14,7 @@ including other versions of pandas.
 
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
--
+- Pandas could not be build on PyPy (GH-42355)
 -
 
 .. ---------------------------------------------------------------------------

--- a/doc/source/whatsnew/v1.3.1.rst
+++ b/doc/source/whatsnew/v1.3.1.rst
@@ -14,7 +14,7 @@ including other versions of pandas.
 
 Fixed regressions
 ~~~~~~~~~~~~~~~~~
-- Pandas could not be build on PyPy (GH-42355)
+- Pandas could not be built on PyPy (:issue:`42355`)
 -
 
 .. ---------------------------------------------------------------------------

--- a/pandas/_libs/src/klib/khash_python.h
+++ b/pandas/_libs/src/klib/khash_python.h
@@ -272,6 +272,8 @@ Py_hash_t PANDAS_INLINE floatobject_hash(PyFloatObject* key) {
 }
 
 
+#define _PandasHASH_IMAG 1000003UL
+
 // replaces _Py_HashDouble with _Pandas_HashDouble
 Py_hash_t PANDAS_INLINE complexobject_hash(PyComplexObject* key) {
     Py_uhash_t realhash = (Py_uhash_t)_Pandas_HashDouble(key->cval.real);
@@ -279,7 +281,7 @@ Py_hash_t PANDAS_INLINE complexobject_hash(PyComplexObject* key) {
     if (realhash == (Py_uhash_t)-1 || imaghash == (Py_uhash_t)-1) {
         return -1;
     }
-    Py_uhash_t combined = realhash + _PyHASH_IMAG * imaghash;
+    Py_uhash_t combined = realhash + _PandasHASH_IMAG * imaghash;
     if (combined == (Py_uhash_t)-1) {
         return -2;
     }

